esrally/mechanic/telemetry.py

```diff
@@ -789,8 +789,6 @@ def __init__(self, metrics_store, node_count_on_host, log_root, node_name):
     def attach_to_node(self, node):
         self.node = node
         self.process = sysstats.setup_process_stats(node.pid)
-
-    def on_benchmark_start(self):
         if self.process is not None:
```

Comment:
You can get rid of this check as it is impossible that `self.process` is `None` at this point.